backend/clubs/views.py

```diff
@@ -4868,27 +4925,47 @@ def cart(self, request, *args, **kwargs):
             return Response(
                 {
                     "tickets": TicketSerializer(cart.tickets.all(), many=True).data,
-                    "sold_out": 0,
+                    "sold_out": [],
                 },
             )
 
-        sold_out_count = 0
+        sold_out_tickets, replacement_tickets = [], []
 
-        replacement_tickets = []
         tickets_in_cart = cart.tickets.values_list("id", flat=True)
+        event_dict = {
+            event["id"]: event["name"]
+            for event in Event.objects.filter(
+                id__in=tickets_to_replace.values_list("event", flat=True).distinct()
+            ).values("id", "name")
+        }
+
+        # Process each ticket type and event
         for ticket_class in tickets_to_replace.values("type", "event").annotate(
             count=Count("id")
         ):
-            # We don't need to lock since we aren't updating holder/owner
-            tickets = Ticket.objects.filter(
-                event=ticket_class["event"],
-                type=ticket_class["type"],
-                owner__isnull=True,
-                holder__isnull=True,
-            ).exclude(id__in=tickets_in_cart)[: ticket_class["count"]]
+            available_tickets = (
+                Ticket.objects.filter(
+                    event=ticket_class["event"],
+                    type=ticket_class["type"],
+                    owner__isnull=True,
+                    holder__isnull=True,
+                )
+                .exclude(id__in=tickets_in_cart)
+                .select_related("event")[: ticket_class["count"]]
+            )
 
-            sold_out_count += ticket_class["count"] - tickets.count()
-            replacement_tickets.extend(list(tickets))
```

Comment:
@joyliu-q I think this is the line that's causing the breaking changes. We're not using the `available_tickets` to replace the ones that have gone stale in the user's cart.